### PR TITLE
quantile regression updates for new hardhat model

### PR DESCRIPTION
**Pull request by topepo**
Depends on hardhat >= 1.4.0.9002. 

The main update is that the hardhat code is standardized on the plural `quantile_levels`.

### CODE REVIEW

**Comment by topepo** [comment]
Closed and reopened to trigger CI for `branch != main`

File: tests/testthat/test-linear_reg_quantreg.R [hunk 1]
```diff
@@ -1,12 +1,15 @@
 test_that('linear quantile regression via quantreg - single quantile', {
   skip_if_not_installed("quantreg")
+  skip_if_not_installed("hardhat", minimum_version = "1.4.0.9002")
+
+  library(hardhat)
```
Comment (hfrick) [inline tests/testthat/test-linear_reg_quantreg.R:5]: Can we avoid that `library()` call here?

File: tests/testthat/_snaps/quantile-reg-specs.md
Comment (hfrick) [inline tests/testthat/_snaps/quantile-reg-specs.md:1]: These snapshots also look like they are meant to test functionality of `hardhat::check_quantile_levels()` so as with the tests in `test-augment.R`, I think we only need one of them.

**Review by hfrick — APPROVED** [review]
Looks good!

**Pull request merged**